addons/delivery/delivery.py

```diff
@@ -228,6 +253,24 @@ def get_price(self, cr, uid, id, order, dt, context=None):
         total = self.pool['res.currency'].compute(cr, uid, order.currency_id.id, order.company_id.currency_id.id, total, context=ctx)
         return self.get_price_from_picking(cr, uid, id, total,weight, volume, quantity, context=context)
 
+    def get_cost_from_picking(self, cr, uid, id, total, weight, volume, quantity, context=None):
```

Comment:
Well, that method is used for calculating price/cost where adding delivery product in sale.order, but also when invoicing stock.picking. 

```get_cost_from_picking``` is not called when invoicing stock.picking but could be if any extra addon adds purchase_price to invoice.line